### PR TITLE
Use publishTo value when it is defined.

### PR DESCRIPTION
**Pull request by hvesalai**
Don't give error about missing repository if publishArtifact is false

### CODE REVIEW

**Comment by hvesalai** [comment]
Ping @dwijnand

**Comment by hvesalai** [comment]
I'll rebase to the correct branch

**Review by dwijnand — DISMISSED** [review]

**Comment by hvesalai** [comment]
The travis failure is from MIMA:

```
method getPublishTo(scala.Option)sbt.librarymanagement.Resolver in object sbt.Classpaths does not have a correspondent in current version

[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("sbt.Classpaths.getPublishTo")
```

**Comment by dwijnand** [comment]
Yeah, you need to create new method.

**Comment by hvesalai** [comment]
done, needs new review and approval

File: main/src/main/scala/sbt/Defaults.scala
Comment (eed3si9n) [inline main/src/main/scala/sbt/Defaults.scala:2342]: `publish / skip` might be better. See discussion here - https://github.com/sbt/sbt/issues/3136

**Review by eed3si9n — CHANGES_REQUESTED** [review]

File: main/src/main/scala/sbt/Defaults.scala
Comment (hvesalai) [reply main/src/main/scala/sbt/Defaults.scala:2342]: Well... I don't mind the new solutoin, but in any case the current code on lines 2342-2344 is clearly a bug and should be replaced soon with some sane alternative.

How about just merging this PR and then working the publish / skip into it as #3136 gets implemented.

Comment (eed3si9n) [reply main/src/main/scala/sbt/Defaults.scala:2342]: publish / skip was implemented here - https://github.com/sbt/sbt/pull/3380

Comment (hvesalai) [reply main/src/main/scala/sbt/Defaults.scala:2342]: So you mean that instead of `publishArtifact.value` we should have here `(skip in publish).value` ?

Comment (eed3si9n) [reply main/src/main/scala/sbt/Defaults.scala:2342]: yes, it should be `!(skip in publish).value`, since you could have published artifacts even if unscoped `publishArtifact.value` returns false.

Comment (eed3si9n) [reply main/src/main/scala/sbt/Defaults.scala:2342]: actually could you hold off on that change until we understand what happened in the other PR please?

Comment (eed3si9n) [reply main/src/main/scala/sbt/Defaults.scala:2342]: I am wondering if a better fix would be to just revert https://github.com/sbt/sbt/pull/3760

Comment (mpollmeier) [reply main/src/main/scala/sbt/Defaults.scala:2342]: responded on https://github.com/sbt/sbt/pull/3760#issuecomment-518797685 - sounds like reverting that one is the way to go

Comment (hvesalai) [reply main/src/main/scala/sbt/Defaults.scala:2342]: #3760 is what broke our build so if that is reverted, all is fine for me.

Comment (hvesalai) [reply main/src/main/scala/sbt/Defaults.scala:2342]: @eed3si9n is it certain that if `skip in publish` is true, then nobody depends on the `publishConfiguration` ?

Comment (eed3si9n) [reply main/src/main/scala/sbt/Defaults.scala:2342]: publish, publishLocal, publishSigned, and sbt-bintray respect the skip in publish, so that should cover the relevant party.

**Comment by eed3si9n** [comment]
Closing this in favor of https://github.com/sbt/sbt/pull/4934

**Pull request closed without merging**
